### PR TITLE
Apply clang-format consistently to the project

### PR DESCRIPTION
**Pull request by alkino**
CI_BRANCHES:NEURON_BRANCH=master

merge after #464

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
```
Comment (pramodk) [inline .clang-format:1]: this file shouldn't be here? (i.e. it will be copied from hpc-coding-convention)

Comment (alkino) [reply .clang-format:1]: it's for ci, this is the file for clang-format 9.0, it is done like that in libsonata, I only copied it from this repository.

File: CMakeLists.txt [hunk 2]
```diff
@@ -44,6 +44,8 @@ list(APPEND CMAKE_MODULE_PATH ${CORENEURON_PROJECT_SOURCE_DIR}/CMake
 # HPC Coding Conventions
 # =============================================================================
 set(CODING_CONV_PREFIX "CORENRN")
+# Disable sort-includes for now, because headers are not clean
```
Comment (tristan0x) [inline CMakeLists.txt:47]: An alternative is to add the file `.clang-format.changes` to the repository with the following change:
```yaml
SortIncludes: false
```

Comment (pramodk) [reply CMakeLists.txt:47]: Tristan's suggestion would be better!

File: .github/workflows/clang_format_check.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+          shell: bash
+          working-directory: ${{runner.workspace}}/CoreNeuron
+          run: |
+              mkdir BUILD && cd BUILD
+              cmake -DCORENRN_CLANG_FORMAT=ON -DCORENRN_ENABLE_MPI=OFF -DCORENRN_ENABLE_OPENMP=OFF ..
+        - name: Run clang-format
+          shell: bash
+          working-directory: ${{runner.workspace}}/CoreNeuron/BUILD
+          run: make VERBOSE=1 check-clang-format
+          env:
```
Comment (alexsavulescu) [inline .github/workflows/clang_format_check.yaml:22]: Seems unnecessary

**Comment by alkino** [comment]
We are trap with: https://bugs.llvm.org/show_bug.cgi?id=36004

**Comment by pramodk** [comment]
@alkino : I will rebase on master. So will push force to this branch.

**Comment by pramodk** [comment]
> We are trap with: https://bugs.llvm.org/show_bug.cgi?id=36004

But, above bug is marked as  `Status: | RESOLVED FIXED` ?

**Comment by pramodk** [comment]
* seems like issue appears (only) when `#pragma` is indented 

```cpp
→ cat ../coreneuron/gpu/nrn_acc_manager.cpp && /usr/local/opt/llvm/bin/clang-format -i --style=file ../coreneuron/gpu/nrn_acc_manager.cpp && cat ../coreneuron/gpu/nrn_acc_manager.cpp
void update_matrix_from_gpu(NrnThread* _nt) {
    if (_nt->compute_gpu && (_nt->end > 0)) {
        // before copying, make sure all computations in the stream are completed

        // clang-format off
        #pragma acc wait(_nt->stream_id a aaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa)
        // clang-format on
    }
}
void update_matrix_from_gpu(NrnThread* _nt) {
    if (_nt->compute_gpu && (_nt->end > 0)) {
// before copying, make sure all computations in the stream are completed

// clang-format off
        #pragma acc wait(_nt->stream_id a aaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa)
        // clang-format on
    }
}
```

* Lets not indent `#pragma`:

```cpp
→ cat ../coreneuron/gpu/nrn_acc_manager.cpp && /usr/local/opt/llvm/bin/clang-format -i --style=file ../coreneuron/gpu/nrn_acc_manager.cpp && cat ../coreneuron/gpu/nrn_acc_manager.cpp
void update_matrix_from_gpu(NrnThread* _nt) {
    if (_nt->compute_gpu && (_nt->end > 0)) {
        // before copying, make sure all computations in the stream are completed

        // clang-format off
#pragma acc wait(_nt->stream_id a aaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa)
        // clang-format on
    }
}
void update_matrix_from_gpu(NrnThread* _nt) {
    if (_nt->compute_gpu && (_nt->end > 0)) {
        // before copying, make sure all computations in the stream are completed

        // clang-format off
#pragma acc wait(_nt->stream_id a aaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa)
        // clang-format on
    }
}
```
now there will be consistent formatting

**Comment by alexsavulescu** [comment]
@pramodk isn't this the same as the fix here: https://github.com/BlueBrain/CoreNeuron/pull/365/commits/c34325134d077d62f35c3887e97c6bd5f6f7c963

**Comment by pramodk** [comment]
* Having extra newline after `clang-format off` avoid this cycle

```cpp
→ /usr/local/opt/llvm/bin/clang-format -i a.cpp  && cat a.cpp
void update_matrix_from_gpu(NrnThread* _nt) {
    if (_nt->compute_gpu && (_nt->end > 0)) {
        // before copying, make sure all computations in the stream are completed

        // clang-format off

        #pragma acc wait(_nt->stream_id a aaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa)
        // clang-format on
    }
}
```

**Comment by pramodk** [comment]
> @pramodk isn't this the same as the fix here: c343251

@alexsavulescu : I completely forgot that! Nico started new PR and I forgot this was already investigated! I will updated accordingly.

**Comment by pramodk** [comment]
For the record, having extra indentation after `// clang-format off` serve the same purpose:

```cpp
/usr/local/opt/llvm/bin/clang-format -i a.cpp  && cat a.cpp
void update_matrix_from_gpu(NrnThread* _nt) {
    if (_nt->compute_gpu && (_nt->end > 0)) {
        // before copying, make sure all computations in the stream are completed

        // clang-format off
            #pragma acc wait(_nt->stream_id a aaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa)
        // clang-format on
    }
}
```

**Comment by pramodk** [comment]
Just a note:

```
# OSX
$ /usr/local/opt/llvm/bin/clang-format --version
clang-format version 11.0.0

# Ubuntu 20.04

$ clang-format-11 --version
Ubuntu clang-format version 11.0.0-2~ubuntu20.04.1

```

and minor diff that they generate:

```
root@7c1316599d1d:/CoreNeuron/build# git diff
diff --git a/coreneuron/mechanism/eion.cpp b/coreneuron/mechanism/eion.cpp
index 2cdc0ec..5b9bb23 100644
--- a/coreneuron/mechanism/eion.cpp
+++ b/coreneuron/mechanism/eion.cpp
@@ -308,7 +308,7 @@ void nrn_cur_ion(NrnThread* nt, Memb_list* ml, int type) {
     for (_iml = 0; _iml < _cntml_actual; ++_iml) {
         pd = ml->data + _iml * nparm;
         ppd = ml->pdata + _iml * 1;
-#elif LAYOUT == 0 /*SoA*/
+#elif LAYOUT == 0         /*SoA*/
     int _cntml_padded = ml->_nodecount_padded;
     pd = ml->data;
     ppd = ml->pdata;
@@ -345,7 +345,7 @@ void nrn_init_ion(NrnThread* nt, Memb_list* ml, int type) {
     for (_iml = 0; _iml < _cntml_actual; ++_iml) {
         pd = ml->data + _iml * nparm;
         ppd = ml->pdata + _iml * 1;
-#elif LAYOUT == 0 /*SoA*/
+#elif LAYOUT == 0         /*SoA*/
     int _cntml_padded = ml->_nodecount_padded;
     pd = ml->data;
     ppd = ml->pdata;
@@ -392,7 +392,7 @@ void second_order_cur(NrnThread* _nt, int secondorder) {
 #if LAYOUT == 1 /*AoS*/
                 for (_iml = 0; _iml < _cntml_actual; ++_iml) {
                     pd = ml->data + _iml * nparm;
-#elif LAYOUT == 0 /*SoA*/
+#elif LAYOUT == 0         /*SoA*/
                 _cntml_padded = ml->_nodecount_padded;
                 pd = ml->data;
                 _PRAGMA_FOR_SEC_ORDER_CUR_ACC_LOOP_
```

**Comment by pramodk** [comment]
retest this please

**Comment by pramodk** [comment]
The branch name of this PR is `clang-format` and I suspect there is package name or variant name somewhere `clang-format` (?) which causing Spack to fail during installation. So I am reopening this PR with different branch name in #465.

**Comment by alkino** [comment]
> > We are trap with: https://bugs.llvm.org/show_bug.cgi?id=36004
> 
> But, above bug is marked as  `Status: | RESOLVED FIXED` ?
> 
> 
> 

Yes, and on version 6, don't ask me why

**Pull request closed without merging**